integrations/whatsapp/src/conversation.ts

```diff
@@ -93,24 +90,26 @@ export async function startConversation(
     },
   })
 
-  const whatsapp = new WhatsAppAPI(ctx.configuration.accessToken)
+  const whatsapp = new WhatsAppAPI({ token: ctx.configuration.accessToken, secure: false })
 
   const language = new Language(templateLanguage)
 
-  const template = new Template(templateName, language, {
-    type: 'body',
-    parameters: templateVariables.map((variable) => ({
-      type: 'text',
-      text: variable,
-    })),
-  })
+  const bodyParams = templateVariables.map((variable) => ({
+    type: 'text',
+    text: variable.toString(),
+  })) as AtLeastOne<BodyParameter>
```

Comment:
**nitpick:** not a fan of this casting because you are possibly casting 2 different things:

1. the structure of a `BodyParameter`
2. the structure of the collection (`BodyParameter[]`-> `AtLeastOne<BodyParameter>`)

And we don't know which one just from reading it

Reply:
Yes, unfortunately the typings this dependency uses are really weird because passing the array as is leads to a type mismatch. I can show you.